pppd/ipcp.c

```diff
@@ -196,6 +196,16 @@ static option_t ipcp_option_list[] = {
       "disable defaultroute option", OPT_ALIAS | OPT_A2CLR,
       &ipcp_wantoptions[0].default_route },
 
+#ifdef __linux__
+    { "replacedefaultroute", o_bool,
+				&ipcp_wantoptions[0].replace_default_route,
+      "Replace default route", OPT_PRIV | 1
+    },
+    { "noreplacedefaultroute", o_bool,
+				&ipcp_allowoptions[0].replace_default_route,
```

Comment:
This is set up so that the options subsystem will clear ipcp_allowoptions[0].replace_default_route if the noreplacedefaultroute option is given; however that won't have any effect. The similar code for the defaultroute option uses OPT_ENABLE to get nodefaultroute to veto defaultroute, but you don't have OPT_ENABLE on the replacedefaultroute option. In any case, replacedefaultroute being privileged means you probably don't want it to be vetoed by a previous noreplacedefaultroute. The change to pppd.8 below says that it would be, though in fact it won't, you probably don't want it to be, and clearing ipcp_allowoptions here is pointless.

Reply:
> replacedefaultroute being privileged means you probably don't want it to be vetoed by a previous noreplacedefaultroute

Mmm, I don't see why?
While I understand why replacedefaultroute should be privileged to avoid e.g. disrupting other work on the system, a user may want to call some ppp network, but avoid affecting the default route even if the usual case is to replace the default route.

Reply:
> > replacedefaultroute being privileged means you probably don't want it to be vetoed by a previous noreplacedefaultroute
> 
> Mmm, I don't see why?
> While I understand why replacedefaultroute should be privileged to avoid e.g. disrupting other work on the system, a user may want to call some ppp network, but avoid affecting the default route even if the usual case is to replace the default route.

What I meant was that you don't want replacedefaultroute to be ineffective because of an earlier noreplacedefaultroute. That is, replacedefaultroute should always set the replace_default_route flag. This is unlike the defaultroute option, which does not work if there has been an earlier nodefaultroute option seen.

Reply:
Ah ok you mean the ordering. I have now fixed it so.